beacon-chain/core/blocks/block_operations.go

```diff
@@ -502,13 +504,13 @@ func ProcessAttesterSlashings(
 		currentEpoch := helpers.SlotToEpoch(beaconState.Slot())
 		var err error
 		var slashedAny bool
-		var val *ethpb.Validator
+		var val *stateTrie.ReadOnlyValidator
 		for _, validatorIndex := range slashableIndices {
-			val, err = beaconState.ValidatorAtIndex(validatorIndex)
```

Comment:
`ValidatorAtIndex` is better here imo. `ProcessAttesterSlashings` from the name itself it is a process function which implies validator object can be updated.  Use `ReadOnly` in this function may be ok for now but in the future when function modifies this can often be overlooked

Reply:
Sure! This change isn't a big improvement so I'll change it back if it has any possible risks.